### DISCUSSION TITLE
Is it possible to std::vector<int> horizontal printing with indent > 0?

### DISCUSSION THREAD

**Discussion by vptyp** [nlohmann/json#4171] [Q&A] [answered]
Hello everyone! 
I think that this is dummy question, but will be glad to hear answer!

So I'm currently trying to write in .json some object in format
```
{
    "name" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9]
}
```
to do this currently using something like

```
string key = "some key";
std::vector<int> v_ = {some values};
 json j;
j[key] = v_;
out << j.dump(4);
```

and result is 

```
        "value": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10
        ]
```

Tried to play with initialization, but no good results.
Somethere saw that vector used with custom types, and with custom to_json function we will get nice results. But I think that with primitive type, initialization working with iteration other vector the same way, as with custom types, but with default to_json function

But, if I use 

```
...
for(auto value:v_)
j[key].push_back(value);
```
result is the same

Currently, don't know how to make it work properly. Thank you for your help!

**Comment by gregmarr** [comment, marked as answer]
There is no way at this time to print an array on a single line.  It has nothing to do with the data itself.

**Reply by vptyp** [reply]
But what's the difference with code snippet in example section? 

```
j["name"] = { 1, 2, 3 };
```

**Reply by gregmarr** [reply]
I don't understand your question.  Can you show your exact code and your exact output?  Your code and output don't line up.

**Reply by vptyp** [reply]
If you will check this [link](https://github.com/nlohmann/json#json-as-first-class-data-type)

you will find this:
```
...
"list": [1, 0, 2],
...
```

```
// add an array that is stored as std::vector (using an initializer list)
j["list"] = { 1, 0, 2 };
```

so I think that this line needs to be changed in docs.

Currently checked this one:
```
    json j;
    std::vector v_ = {1, 2, 3};
    j["v_"] = v_;
    j["init"] = {1, 2, 3};
    std::ofstream out;
    out.open("path.json");
    out << j.dump(4) << std::endl;
    out.close();
```

both:
```
{
    "init": [
        1,
        2,
        3
    ],
    "v_": [
        1,
        2,
        3
    ]
}
```

**Reply by vptyp** [reply]
Really no way to print array in single line

Is it make sense to write specific serialization methods for primitive type arrays?
